include/acquire.zarr.h

```diff
@@ -0,0 +1,274 @@
+     * @param settings The Zarr stream settings struct.
+     * @param index The index of the dimension to get.
+     * @return The properties for the @p index th dimension, or an uninitialized struct on
+     * failure.
+     */
+    ZarrDimensionProperties ZarrStreamSettings_get_dimension(
+      const ZarrStreamSettings* settings,
+      size_t index);
+
+    /**
+     * @brief Get the multiscale flag for the Zarr stream.
+     * @param settings The Zarr stream settings struct.
+     * @return The multiscale flag for the Zarr stream, or false on failure.
+     */
+    bool ZarrStreamSettings_get_multiscale(const ZarrStreamSettings* settings);
+
+    /**
+     * @brief Get the JSON-formatted custom metadata for the Zarr stream.
+     * @param settings The Zarr stream settings struct.
+     * @return The JSON-formatted custom metadata for the Zarr stream, or NULL on
+     * failure.
+     */
+    const char* ZarrStreamSettings_get_custom_metadata(
+      const ZarrStreamSettings* settings);
+
+    /**
+     * @brief Create a Zarr stream.
+     * @param[in, out] settings The settings for the Zarr stream.
+     * @param[in] version The version of the Zarr stream. 2 or 3.
+     * @return A pointer to the Zarr stream struct, or NULL on failure.
+     */
+    ZarrStream* ZarrStream_create(ZarrStreamSettings* settings,
+                                  ZarrVersion version);
+
+    /**
+     * @brief Destroy a Zarr stream.
+     * @details This function frees the memory allocated for the Zarr stream.
+     * @param stream The Zarr stream struct to destroy.
+     */
+    void ZarrStream_destroy(ZarrStream* stream);
+
+    /**
+     * @brief Append data to the Zarr stream.
+     * @details This function will block while chunks are compressed and written
+     * to the store. It will return when all data has been written.
+     * @param[in, out] stream The Zarr stream struct.
+     * @param[in] data The data to append.
+     * @param[in] bytes_in The number of bytes in @p data. It should be at least
+     * the size of a single frame.
+     * @param[out] bytes_out The number of bytes written to the stream.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus ZarrStream_append(ZarrStream* stream,
+                                 const void* data,
+                                 size_t bytes_in,
+                                 size_t* bytes_out);
+
+    /**
+     * @brief Get the version (i.e., 2 or 3) of the Zarr stream.
+     * @param stream The Zarr stream struct.
+     * @return The version of the Zarr stream.
+     */
+    ZarrVersion ZarrStream_get_version(const ZarrStream* stream);
+
+    /**
+     * @brief Get a copy of the settings for the Zarr stream.
+     * @param stream The Zarr stream struct.
+     * @return A copy of the settings for the Zarr stream.
+     */
+    ZarrStreamSettings* ZarrStream_get_settings(const ZarrStream* stream);
```

Comment:
This relates to my earlier question about `ZarrStreamSettings_copy`. Is there a specific use case for copying stream settings objects?